analysis/shaman/src/shadowlands/conduits/TumblingWaves.tsx

```diff
@@ -0,0 +1,29 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+
+class TumblingWaves extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  protected spellUsable!: SpellUsable;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasConduitBySpellID(SPELLS.TUMBLING_WAVES_CONDUIT.id);
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.TUMBLING_WAVES_BUFF),
+      this.onApplyBuff,
+    );
+  }
+
+  onApplyBuff() {
+    this.spellUsable.endCooldown(SPELLS.PRIMORDIAL_WAVE_CAST.id);
```

Comment:
this can throw if (for some reason) the spell isn't on cooldown. not familiar with shaman so dunno if that is even possible, but wanted to confirm that you're aware

(throwing an error here won't crash the page, it will just disable this module and dependents)

Reply:
Good point, you never know what blizzard changes.